sdk/storage/azure-storage-blob/src/test/java/com/azure/android/storage/blob/BlockBlobTest.java

```diff
@@ -0,0 +1,427 @@
+        };
+    }
+
+    // TODO: (gapra) Add iftags
+    @DataProvider
+    public static Object[][] accessConditionsSuccess() {
+        return new Object[][] {
+            {null,    null,    null,         null},       // 0
+            {oldDate, null,    null,         null},       // 1
+            {null,    newDate, null,         null},       // 2
+            {null,    null,    receivedEtag, null},       // 3
+            {null,    null,    null,         garbageEtag} // 4
+        };
+    }
+
+    // TODO: (gapra) Add iftags
+    @DataProvider
+    public static Object[][] accessConditionsFail() {
+        return new Object[][] {
+            {newDate, null,    null,        null},        // 0
+            {null,    oldDate, null,        null},        // 1
+            {null,    null,    garbageEtag, null},        // 2
+            {null,    null,    null,        receivedEtag} // 3
+        };
+    }
+
+    @BeforeClass
+    public static void setupClass() {
+        asyncClient = initializeDefaultAsyncBlobClientBuilder(enableFiddler()).build();
+        syncClient = initializeDefaultSyncBlobClientBuilder(enableFiddler()).build();
+    }
+
+    @Before
+    public void setupTest() {
+        // Create container
+        containerName = generateResourceName();
+        syncClient.createContainer(containerName);
+
+        // Create blob
+        blobName = generateResourceName();
+        String blockId = generateBlockID();
+        syncClient.stageBlock(containerName, blobName, blockId, getDefaultData(), null);
+        List<String> blockIds = new ArrayList<>();
+        blockIds.add(blockId);
+        syncClient.commitBlockList(containerName, blobName, blockIds, false);
+    }
+
+    @After
+    public void teardownTest() {
+        syncClient.deleteContainer(containerName);
+    }
+
+    @Test
+    public void stageBlock() {
+        // Setup
+        String blockId = generateBlockID();
+
+        // When
+        BlockBlobsStageBlockResponse response = syncClient.stageBlockWithRestResponse(containerName, blobName, blockId, getDefaultData(), null, null, null ,null ,null ,null, null);
+
+        // Then
+        assertEquals(201, response.getStatusCode());
+        BlockBlobStageBlockHeaders headers = response.getDeserializedHeaders();
+        assertNotNull(headers.getXMsContentCrc64());
+        assertNotNull(headers.getRequestId());
+        assertNotNull(headers.getVersion());
+        assertNotNull(headers.getDateProperty());
+        assertTrue(headers.isServerEncrypted() != null && headers.isServerEncrypted());
+    }
+
+    @Test
+    public void stageBlockMin() {
+        // Setup
+        String blockId = generateBlockID();
+
+        // When
+        Void response = syncClient.stageBlock(containerName, blobName, blockId, getDefaultData(), null);
+
+        // Then
+        // When list block support added, check blockBlobClient.listBlocks(BlockListType.ALL).getUncommittedBlocks().size() == 1
```

Comment:
You could also maybe do a commit block list since it'll 400 or something if the block wasn't staged.

Reply:
Ah yeah